src/decoder/lattice-faster-decoder.cc

```diff
@@ -715,7 +717,7 @@ BaseFloat LatticeFasterDecoder::ProcessEmitting(DecodableInterface *decodable) {
     StateId state = best_elem->key;
     Token *tok = best_elem->val;
     cost_offset = - tok->tot_cost;
-    for (fst::ArcIterator<fst::Fst<Arc> > aiter(fst_, state);
+    for (fst::ArcIterator<Fst > aiter(dynamic_cast<Fst &>(const_cast<fst::Fst<Arc> &>(fst_)), state);
```

Comment:
figure out a way to reduce this line length, I suggest declaring a reference variable as follows, at the top of this function:
`FstType &fst = dynamic_cast<FstType&>(const_cast<fst::Fst<Arc> &>(fst_));`
This will also save a lot of vtable lookups, since dynamic_cast is a nontrivial operation.